### PR TITLE
Feature/gap 2152 get email from user service

### PR DESCRIPTION
**Pull request by ryan-tco**
## Description

Adds an endpoint that find-be can use to acquire oneLogin email addresses to use for notifications 

Ticket # and link

GAP 2152

Summary of the changes and the related issue. List any dependencies that are required for this change:

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [TODO ] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

### CODE REVIEW

File: src/main/java/gov/cabinetofice/gapuserservice/dto/UserEmailDto.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package gov.cabinetofice.gapuserservice.dto;
+
+import lombok.Builder;
+import lombok.Data;
+
+
+@Data
+@Builder
+public class UserEmailDto {
```
Comment (john-tco) [inline src/main/java/gov/cabinetofice/gapuserservice/dto/UserEmailDto.java:9]: Should we use a `record` here?

Comment (ryan-tco) [reply src/main/java/gov/cabinetofice/gapuserservice/dto/UserEmailDto.java:9]: I don't believe so as we never need to map this back into something we're storing into DB

Comment (dominicwest) [reply src/main/java/gov/cabinetofice/gapuserservice/dto/UserEmailDto.java:9]: `record` is a good shout, just a cleaner way of defining a DTO. But both solutions will of course work fine

**Review by dominicwest — APPROVED** [review]

**Review by dominicwest — APPROVED** [review]

File: src/main/java/gov/cabinetofice/gapuserservice/service/SecretAuthService.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package gov.cabinetofice.gapuserservice.service;
 
 import gov.cabinetofice.gapuserservice.exceptions.UnauthorizedException;
+import gov.cabinetofice.gapuserservice.service.encryption.AwsEncryptionServiceImpl;
```
Comment (dylanwrightCO) [inline src/main/java/gov/cabinetofice/gapuserservice/service/SecretAuthService.java:4]: Can remove this unused import

**Review by dylanwrightCO — DISMISSED** [review]

File: src/main/java/gov/cabinetofice/gapuserservice/web/UserController.java [hunk 3]
```diff
@@ -150,5 +151,11 @@ public ResponseEntity<String> deleteUser(HttpServletRequest httpRequest, @PathVa
         return ResponseEntity.ok("success");
 
     }
+
+    @PostMapping("/users/emails")
```
Comment (dominicwest) [inline src/main/java/gov/cabinetofice/gapuserservice/web/UserController.java:155]: Endpoint was still requiring jwt auth - pushed a fix

**Review by dominicwest — APPROVED** [review]

**Pull request merged**